openmls/src/group/mls_group/updates.rs

```diff
@@ -145,19 +145,21 @@ impl MlsGroup {
 
         // Here we clone our own leaf to rekey it such that we don't change the
         // tree.
-        // The new leaf node will be applied later when the proposal is commited.
+        // The new leaf node will be applied later when the proposal is committed.
         let mut rekeyed_own_leaf = tree
             .own_leaf_node()
             .ok_or_else(|| LibraryError::custom("The tree is broken. Couldn't find own leaf."))?
             .clone();
-        if let Some(key_package) = key_package {
+        if let Some(leaf) = leaf_node {
             let private_key: Vec<u8> = backend
                 .key_store()
-                .read(key_package.hpke_init_key().as_slice())
+                .read(&LeafNode::encryption_key_label(
```

Comment:
Is there already documentation on what we store under which label? Since this is all somewhat hidden due to the "global" nature of the key store, it might make sense to create an overview somewhere.